src/app/client/src/app/modules/shared-feature/components/collection-player/collection-player.component.ts

```diff
@@ -601,6 +605,35 @@ export class CollectionPlayerComponent implements OnInit, OnDestroy, AfterViewIn
     });
   }
 
+  getChildren(collection) {
```

Comment:
This will end up in infinite loop if there is any issue in code. We should not write calling the same function by itself based on condition.

Reply:
Without recursive function i won't get children data to filter those fields

Reply:
and it won't end up infinite loop, since we are returning  data